pkg/sql/backoff_manager.go

```diff
@@ -0,0 +1,53 @@
+package sql
+
+import (
+	"github.com/ThreeDotsLabs/watermill"
+	"strings"
+	"time"
+)
+
+// BackoffManager handles errors or empty result sets and compute a backoff and a resulting error.
+// You could for example create a stateful version that compute backoffs depending on the error frequency or make errors more or less persistent
+type BackoffManager interface {
+	// Compute computes the backoff time depending on the error or the absence of result.
+	Compute(logger watermill.LoggerAdapter, noMsg bool, err error) time.Duration
+}
+
+func NewDefaultBackoffManager(pollInterval, retryInterval time.Duration) BackoffManager {
+	if pollInterval == 0 {
+		pollInterval = time.Second
+	}
+	if retryInterval == 0 {
+		retryInterval = time.Second
+	}
+	return &defaultBackoffManager{
+		retryInterval: retryInterval,
+		pollInterval:  pollInterval,
+	}
+}
+
+type defaultBackoffManager struct {
+	pollInterval  time.Duration
+	retryInterval time.Duration
+}
+
+func (d defaultBackoffManager) Compute(logger watermill.LoggerAdapter, noMsg bool, err error) time.Duration {
```

Comment:
I like the idea overall! I'm a bit concerned there's a lot happening in the `Compute` method. It computes something, but also logs messages. Maybe we could name it more clearly, perhaps something related with error handling?

Reply:
`HandleError` ?